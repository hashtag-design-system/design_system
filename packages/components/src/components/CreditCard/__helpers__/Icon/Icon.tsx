import React from "react";

type Props = {};

export const Icon: React.FC<Props> = () => {
  return (
    <svg
      width={28}
      height={28}
      className="icon credit-card__icon"
      data-testid="icon"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M6.472 4C4.036 4 2 6.037 2 8.472v7.98c0 2.434 2.037 4.471 4.472 4.471h11.057c2.434 0 4.471-2.037 4.471-4.471v-7.98C22 6.038 19.963 4 17.529 4H6.472zm0 1.538h11.057c1.566 0 2.933 1.367 2.933 2.934v.913h-4.616a.767.767 0 01-.77-.77c0-.427.343-.769.77-.769a.77.77 0 000-1.538 2.32 2.32 0 00-2.307 2.307c0 .994.649 1.842 1.538 2.164v3.822c-.922.626-1.538 1.677-1.538 2.86 0 .71.225 1.37.6 1.924H9.862a3.414 3.414 0 00.6-1.924c0-1.182-.617-2.234-1.539-2.86v-3.823c.89-.321 1.538-1.169 1.538-2.163a2.32 2.32 0 00-2.307-2.307H6.615a.771.771 0 00-.072 0 .769.769 0 00-.071 0 .773.773 0 00.143 1.538h1.539c.427 0 .77.342.77.77 0 .426-.343.769-.77.769H3.538V8.47c0-1.566 1.367-2.933 2.934-2.933zm-2.934 5.385h3.847V14H3.538v-3.077zm13.077 0h3.847V14h-3.847v-3.077zM3.538 15.54H7c1.03 0 1.923.893 1.923 1.923S8.03 19.385 7 19.385h-.53c-1.565 0-2.932-1.367-2.932-2.934v-.912zm13.462 0h3.462v.913c0 1.566-1.367 2.933-2.933 2.933H17c-1.03 0-1.923-.894-1.923-1.924 0-1.03.893-1.922 1.923-1.922z" fill="var(--grey-1)" />
    </svg>
  );
};
