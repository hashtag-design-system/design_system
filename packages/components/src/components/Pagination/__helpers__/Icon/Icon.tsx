import React from "react";

type FProps = Omit<React.SVGProps<SVGSVGElement>, "className">;

export const Icon: React.FunctionComponent<FProps> = ({ d, ...props }) => {
  return (
    <svg
      className="icon pagination__btn__icon"
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      data-testid="icon"
      {...props}
    >
      <path d={d} strokeWidth={1.5} strokeLinecap="round" strokeLinejoin="round" />
    </svg>
  );
};
