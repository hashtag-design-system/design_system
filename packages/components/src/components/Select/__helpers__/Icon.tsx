import { motion, SVGMotionProps } from "framer-motion";
import React from "react";

export const Icon: React.FC<{ clicked: boolean; onClick?: SVGMotionProps<SVGElement>["onClick"] }> = ({ clicked, ...props }) => {
  return (
    <motion.svg
      width={24}
      height={24}
      fill="none"
      className="input__icon select__icon"
      animate={clicked ? { rotate: 180 } : { rotate: 0 }}
      // * A "spring" animation might be too much and useless for this element
      {...props}
    >
      <path d="M21 7l-9 10L3 7" stroke="var(--grey-7)" strokeWidth={2} strokeLinecap="round" strokeLinejoin="round" />
    </motion.svg>
  );
};

export default Icon;
