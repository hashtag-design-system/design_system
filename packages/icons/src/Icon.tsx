import Arrow from "./Arrow";
import { Calendar } from "./Calendar";
import { CheckMark, CheckMarkCircle } from "./CheckMark";
import Chevron from "./Chevron";
import { Star } from "./Star";
import { Resize } from "./Resize";
import { Password } from "./Password";
import { Forbidden } from "./Forbidden";
import { Bookmark } from "./Bookmark";
import { Flame } from "./Flame";
import { Edit, Pencil } from "./Edit";
import { SwimmingPool } from "./SwimmingPool";
import { Beach, BeachBall, BeachChair, BeachUmbrella } from "./Beach";
import { Sunbed, SunbedWithMatress } from "./Sunbed";
import { TrashBin } from "./TrashBin";
import { Close, CloseCircle } from "./Close";
import Files from "./Files";
import Math from "./Math";
import Navigation from "./Navigation";
import Notification from "./Notification";
import Payment from "./Payment";
import People from "./People";
import Search from "./Search";
import Social from "./Social";
import Logo from "./Logo";

type SubComponents = {
  Chevron: typeof Chevron;
  Arrow: typeof Arrow;
  Math: typeof Math;
  Calendar: typeof Calendar;
  Files: typeof Files;
  Logo: typeof Logo;
  Navigation: typeof Navigation;
  Notification: typeof Notification;
  Payment: typeof Payment;
  People: typeof People;
  Search: typeof Search;
  Social: typeof Social;
  Close: typeof Close;
  CloseCircle: typeof CloseCircle;
  CheckMark: typeof CheckMark;
  CheckMarkCircle: typeof CheckMarkCircle;
  Star: typeof Star;
  Resize: typeof Resize;
  Password: typeof Password;
  Forbidden: typeof Forbidden;
  Bookmark: typeof Bookmark;
  Flame: typeof Flame;
  Pencil: typeof Pencil;
  Edit: typeof Edit;
  SwimmingPool: typeof SwimmingPool;
  Beach: typeof Beach;
  BeachBall: typeof BeachBall;
  BeachChair: typeof BeachChair;
  BeachUmbrella: typeof BeachUmbrella;
  Sunbed: typeof Sunbed;
  SunbedWithMatress: typeof SunbedWithMatress;
  TrashBin: typeof TrashBin;
};

type IconProps = {};

export const Icon: React.FC<IconProps> & SubComponents = () => {
  return <div></div>;
};

Icon.displayName = "Icon";

Icon.Chevron = Chevron;
Icon.Arrow = Arrow;
Icon.Math = Math;
Icon.Calendar = Calendar;
Icon.Files = Files;
Icon.Logo = Logo;
Icon.Navigation = Navigation;
Icon.Navigation = Navigation;
Icon.Notification = Notification;
Icon.Payment = Payment;
Icon.People = People;
Icon.Search = Search;
Icon.Social = Social;
Icon.Close = Close;
Icon.CloseCircle = CloseCircle;
Icon.CheckMark = CheckMark;
Icon.CheckMarkCircle = CheckMarkCircle;
Icon.Star = Star;
Icon.Resize = Resize;
Icon.Password = Password;
Icon.Forbidden = Forbidden;
Icon.Bookmark = Bookmark;
Icon.Flame = Flame;
Icon.Pencil = Pencil;
Icon.Edit = Edit;
Icon.SwimmingPool = SwimmingPool;
Icon.Beach = Beach;
Icon.BeachBall = BeachBall;
Icon.BeachChair = BeachChair;
Icon.BeachUmbrella = BeachUmbrella;
Icon.Sunbed = Sunbed;
Icon.SunbedWithMatress = SunbedWithMatress;
Icon.TrashBin = TrashBin;

export default Icon;
