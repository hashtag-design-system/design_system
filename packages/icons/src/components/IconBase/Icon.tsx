import React from "react";

type cssColorAttribute = React.SVGAttributes<SVGElement>["color"];

export type Props = {
  d: string;
  size?: number;
  width?: number;
  height?: number;
  viewBox?: React.SVGAttributes<SVGElement>["viewBox"];
  fill?: cssColorAttribute;
  stroke?: cssColorAttribute;
  strokeWidth?: number;
  strokeLinecap?: React.SVGAttributes<SVGElement>["strokeLinecap"];
};

const Icon: React.FC<Props & React.SVGAttributes<SVGElement>> = ({
  d,
  size = 24,
  width,
  height,
  viewBox = "0 0 24 24",
  fill,
  stroke = "#999999",
  strokeWidth = 3,
  strokeLinecap = "round",
  ...props
}) => {
  const { className, ...rest } = props;
  // let classNames = addClassnames(`input input-multiline`, props);
  // Size is to set both the width and the height, without duplicating props and their values
  if (size) {
    width = size;
    height = size;
  }

  return (
    <svg
      // className={classNames}
      viewBox={viewBox}
      width={`${width}px`}
      height={`${height}px`}
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
      strokeWidth={strokeWidth}
      strokeLinecap={strokeLinecap}
      {...rest}
    >
      <path fill={fill} d={d} />
    </svg>
  );
};

export default Icon;
