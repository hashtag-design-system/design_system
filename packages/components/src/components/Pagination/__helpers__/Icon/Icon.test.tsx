import { render, screen } from "@testing-library/react";
import { Icon } from "./Icon";

describe("Pagination <Icon />", () => {
  test("default behaviour", () => {
    render(<Icon />);
    const icon = screen.getByTestId("icon");

    expect(icon).toBeVisible();
    expect(icon).toHaveAttribute("class");
    expect(icon).toHaveClass("pagination__btn__icon");
    expect(icon).toHaveAttribute("width", "1em");
    expect(icon).toHaveAttribute("height", "1em");
    expect(icon).toHaveAttribute("viewBox", "0 0 24 24");
    expect(icon).toHaveAttribute("fill");
    expect(icon).toHaveAttribute("xmlns");
    expect(icon).toMatchSnapshot();
    expect(icon.tagName.toLowerCase()).toBe("svg")
    expect(icon.children).toHaveLength(1);

    const child = icon.children[0];
    expect(child.tagName.toLowerCase()).toBe("path");
    expect(child).toHaveAttribute("stroke-width", "1.5");
    expect(child).toHaveAttribute("stroke-linecap", "round");
    expect(child).toHaveAttribute("stroke-linejoin", "round");
    expect(child).not.toHaveAttribute("d");
    expect(child.children).toHaveLength(0);
  });
  test("with d", () => {
    render(<Icon d="M17 21L7 12l10-9" />);
    const icon = screen.getByTestId("icon");

    expect(icon).toMatchSnapshot();

    const child = icon.children[0];
    expect(child).toHaveAttribute("d", "M17 21L7 12l10-9");
  });
});
