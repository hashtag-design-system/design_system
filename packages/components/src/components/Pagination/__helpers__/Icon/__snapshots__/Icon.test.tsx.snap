// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination <Icon /> default behaviour 1`] = `
<svg
  class="icon pagination__btn__icon"
  data-testid="icon"
  fill="none"
  height="1em"
  viewBox="0 0 24 24"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
  />
</svg>
`;

exports[`Pagination <Icon /> with d 1`] = `
<svg
  class="icon pagination__btn__icon"
  data-testid="icon"
  fill="none"
  height="1em"
  viewBox="0 0 24 24"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M17 21L7 12l10-9"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
  />
</svg>
`;
